packages/frontend/src/components/copy-button.tsx

```diff
@@ -1,20 +1,52 @@
 'use client'
 
-import { useState } from 'react'
+import { type ReactNode, useState } from 'react'
 
 import { useCopyToClipboard } from '~/hooks/use-copy-to-clipboard'
 import { useTimeout } from '~/hooks/use-timeout'
 import { CopyIcon } from '~/icons/copy'
+import { CopyBookIcon } from '~/icons/copy-book'
 import { SatisfiedIcon } from '~/icons/satisfied'
 import { cn } from '~/utils/cn'
 import { Tooltip, TooltipContent, TooltipTrigger } from './core/tooltip/tooltip'
 
-interface Props {
+interface CopyButtonProps {
   toCopy: string
   className?: string
+  variant?: 'default' | 'glossary'
 }
 
-export function CopyButton(props: Props) {
+interface IconConfig {
+  default: {
+    active: ReactNode
+    inactive: ReactNode
+  }
+  glossary: {
+    active: ReactNode
+    inactive: ReactNode
+  }
+}
+
+const iconConfig: IconConfig = {
+  default: {
+    active: <SatisfiedIcon className="size-4" />,
+    inactive: <CopyIcon className="size-4 fill-gray-750 dark:fill-white/80" />,
+  },
+  glossary: {
+    active: <SatisfiedIcon className="fill-green-700  dark:fill-green-450" />,
+    inactive: (
+      <CopyBookIcon className="fill-blue-700 hover:fill-blue-600 dark:fill-blue-500 dark:hover:fill-blue-550" />
+    ),
+  },
+}
```

Comment:
i don't like the idea of having two different styles for copy button, let's use one

Reply:
https://github.com/l2beat/l2beat/pull/6017/commits/4e81a041ca5ea8c1224b4cf05ad37cfdc1174ed2